internal/services/automation/automation_software_update_configuration.go

```diff
@@ -0,0 +1,949 @@
+package automation
+
+import (
+	"context"
+	"fmt"
+	"regexp"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/services/preview/automation/mgmt/2020-01-13-preview/automation"
+	"github.com/Azure/go-autorest/autorest/date"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/sdk"
+	validate3 "github.com/hashicorp/terraform-provider-azurerm/internal/services/apimanagement/validate"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/automation/parse"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/automation/validate"
+	computeValidate "github.com/hashicorp/terraform-provider-azurerm/internal/services/compute/validate"
+	validate2 "github.com/hashicorp/terraform-provider-azurerm/internal/services/resource/validate"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/suppress"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
+	"github.com/hashicorp/terraform-provider-azurerm/utils"
+)
+
+func parseTime(s string) *date.Time {
+	t, _ := time.Parse(time.RFC3339, s)
+	return &date.Time{
+		Time: t,
+	}
+}
+
+func timeString(t *date.Time) string {
+	if t == nil {
+		return ""
+	}
+	return t.Time.Format(time.RFC3339)
+}
+
+func stringSlice(input *[]string) []string {
+	if input == nil {
+		return nil
+	}
+	return *input
+}
+
+func deFloat64(ptr *float64) float64 {
+	if ptr == nil {
+		return 0
+	}
+	return *ptr
+}
+
+type Tag struct {
+	Tag    string   `tfschema:"tag"`
+	Values []string `tfschema:"values"`
+}
+
+type AzureQuery struct {
+	Scope     []string `tfschema:"scope"`
+	Locations []string `tfschema:"locations"`
+	Tags      []Tag    `tfschema:"tags"`
+	TagFilter string   `tfschema:"tag_filter"`
+}
+
+func (a *AzureQuery) LoadSDKTags(tags map[string][]string) {
+	if tags == nil {
+		return
+	}
+	for k, vs := range tags {
+		t := Tag{}
+		t.Tag = k
+		for _, v := range vs {
+			t.Values = append(t.Values, v)
+		}
+		a.Tags = append(a.Tags, t)
+	}
+}
+func (a *AzureQuery) ToSDKTags() map[string][]string {
+	if len(a.Tags) == 0 {
+		return nil
+	}
+	m := map[string][]string{}
+	for _, tag := range a.Tags {
+		m[tag.Tag] = tag.Values
+	}
+	return m
+}
```

Comment:
and not using the builtin tags?

Reply:
the build tags is a map of string to string, in this request is a map of string to a slice of string.